src/inference/src/dev/core_impl.cpp

```diff
@@ -1473,14 +1475,15 @@ ov::SoPtr<ov::ICompiledModel> ov::CoreImpl::load_model_from_cache(
                 update_config[ov::loaded_from_cache.name()] = true;
 
                 if (util::contains(plugin.get_property(ov::supported_properties), ov::weights_path)) {
-                    std::string weights_path = cacheContent.modelPath;
-                    auto pos = weights_path.rfind('.');
-                    if (pos != weights_path.npos && weights_path.substr(pos) == ".xml") {
-                        weights_path = weights_path.substr(0, pos);
-                        weights_path += ".bin";
+                    std::filesystem::path weights_path;
+                    if (cacheContent.model != nullptr && cacheContent.model->has_rt_info("__weights_path")) {
+                        weights_path = cacheContent.model->get_rt_info().at("__weights_path").as<std::string>();
+                    } else {
+                        weights_path = cacheContent.modelPath;
+                        weights_path.replace_extension(".bin");
                     }
-                    if (ov::util::file_exists(weights_path)) {
```

Comment:
This should not be required as our util will work with filesystem::path

Reply:
I know, but since we are working with std API, then no utils are required.

Reply:
The `ov::util::file_exists` works little different then std API. Our util return false and no throw, this version of std may throw so flow cane little different. 
Our util just wrap it but if throw it return false.